modules/tinymce/src/themes/silver/main/ts/ButtonState.ts

```diff
@@ -0,0 +1,44 @@
+import { Behaviour, Disabling, Receiving } from '@ephox/alloy';
+import { Arr } from '@ephox/katamari';
+
+import Editor from 'tinymce/core/api/Editor';
+
+import { ReadyUiReferences } from './modes/UiReferences';
+
+export const ButtonStateChannel = 'silver.buttonstate';
+
+const broadcastEvents = (uiRefs: ReadyUiReferences): void => {
+  const motherships = [ uiRefs.mainUi.mothership, ...uiRefs.uiMotherships ];
+  Arr.each(motherships, (m) => {
+    m.broadcastOn([ ButtonStateChannel ], '');
+  });
+};
+
+const setupEventsForButton = (editor: Editor, uiRefs: ReadyUiReferences): void => {
+  editor.on('init', () => {
+    broadcastEvents(uiRefs);
+  });
+
+  editor.on('SwitchMode', () => {
+    broadcastEvents(uiRefs);
+  });
+
+  editor.on('NodeChange', () => {
+    broadcastEvents(uiRefs);
+  });
+};
+
+const toggleButtonStateOnReceive = (shouldDisable: () => boolean): Behaviour.NamedConfiguredBehaviour<any, any> => Receiving.config({
```

Comment:
```suggestion
const buttonStateToggleOnReceive = (shouldDisable: () => boolean): Behaviour.NamedConfiguredBehaviour<any, any> => Receiving.config({
```
`toggleButtonStateOnReceive` sounded in reference to "Toggle Button" to me. Suggestion is just opinion.

Reply:
Maybe just `ButtonState.toggleOnReceive` since we are importing `ButtonState` module and to call that it would be `ButtonState.toggleOnReceive`? Would that be clearer?

Reply:
Sounds good to me 👍